interfaces/cython/cantera/test/test_composite.py

```diff
@@ -145,6 +145,53 @@ def test_collect_data(self):
         self.assertIn('X', collected)
         self.assertEqual(collected['X'].shape, (0, self.gas.n_species))
 
+    def test_append_no_norm_data(self):
+        gas = ct.Solution("h2o2.yaml")
+        gas.TP = 300, ct.one_atm
+        gas.X = gas.X - 1.e-16
```

Comment:
By default, this will normalize the mole fractions and you'll end up with a sum equal to 1.0 still. Then, setting `normalize=False` in the `append()` below doesn't actually do anything.

As another note, changing the mole fractions by 1E-16 here and elsewhere is not a big enough change. 1E-16 is approximately at the precision of a double floating point number, so this change may or may not actually be changing the numbers in the array. It'd be a more stringent test to set all the un-normalized mole fractions to 0.1 or something, since the phase has more than 10 species. That way you're sure that everything is working properly. Apologies that I didn't notice this before...

Reply:
Actually, I think setting a value of 0.3 (or something else that won't sum to 1.0) for all the mole fractions would be more robust.